src/git.py

```diff
@@ -155,16 +155,14 @@ def _cleanup_git_dir(self) -> None:
         shutil.rmtree(self._git_dir)
 
     # Pass in the dao so that patches can be updated when they are pushed, this way less lost data when an error happens
-    def apply_patchset_and_cleanup(self, patchset: Patchset, messages_dao: message_dao.MessageDao):
+    def apply_patchset_and_cleanup(self, patchset: Patchset, message: Message, message_dao: message_dao.MessageDao):
```

Comment:
I don't think I understand how this new message parameter works. Can you add a comment to this function that explains what the parameters do?

Reply:
I'll add a comment, but the gist of it is that, previously, we'd utilize the patch's message id to look up the message in the database. However, that lookup is unnecessary as we already have these messages stored prior to calling this method. Thus, it's more efficient to simply pass these messages into the method.

Reply:
Added a comment to explain what the parameter does.